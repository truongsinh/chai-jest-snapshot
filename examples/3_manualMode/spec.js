const expect = require("chai").expect;

describe("using chaiJestSnapshot in manual mode", function() {
  it("takes the snapshot filename and name as arguments", function() {
    expect({ foo: "bar" }).to.matchSnapshot(__filename + ".snap", "snapshot 1");
  });

  it("uses an optional third argument to update a snapshot", function() {
    expect({ foo: "not bar" }).to.matchSnapshot(__filename + ".snap", "snapshot 2", true);
  });

  it("can use an environment variable to update the snapshot instead", function() {
    process.env.CHAI_JEST_SNAPSHOT_UPDATE_ALL = "true";
    expect({ foo: "something else" }).to.matchSnapshot(__filename + ".snap", "snapshot 3");
  });
});
