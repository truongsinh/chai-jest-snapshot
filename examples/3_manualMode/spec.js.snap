exports[`snapshot 1`] = `
Object {
  "foo": "bar"
}
`;

exports[`snapshot 2`] = `
Object {
  "foo": "not bar"
}
`;

exports[`snapshot 3`] = `
Object {
  "foo": "something else"
}
`;
